packages/admin/cms-admin/src/dam/DataGrid/FolderDataGrid.tsx

```diff
@@ -264,10 +276,17 @@ const FolderDataGrid = ({
             hideHoverStyles();
             hideUploadFooter();
         },
-        onDrop: async (acceptedFiles: File[], fileRejections: FileRejection[]) => {
+        onDrop: async (acceptedFiles: File[], fileRejections: FileRejection[], event) => {
             hideHoverStyles();
             hideUploadFooter();
+
             await fileUploadApi.uploadFiles({ acceptedFiles, fileRejections }, currentFolderId);
+
+            // react-dropzone doesn't support folder drops natively
+            // the only way to find out if an empty folder was dropped is if there are no rejected files and no accepted files
+            if (!fileRejections.length && !acceptedFiles.length) {
+                snackbarApi.showSnackbar(emptyFolderSnackbarElement);
```

Comment:
Please "create" the element directly here:

```tsx
snackbarApi.showSnackbar(
<Snackbar
  anchorOrigin={{ vertical: "bottom", horizontal: "left" }}
  autoHideDuration={5000}
  TransitionComponent={(props: SlideProps) => <Slide {...props} direction="right" />}
  message={<FormattedMessage id="comet.dam.upload.noEmptyFolders" defaultMessage={"Empty folders can't be uploaded"} />}
/>
)
```

Reply:
Why is the slight in direction from the right? Should we change this for all Snackbars?

Reply:
Why create the element there? It makes it less readable imo 🤔

Reply:
> Why create the element there? It makes it less readable imo 🤔

I prefer to create elements as close to where they are used. When reading the code I don't have to "jump" to another place to understand it. But you're right, this is an opinionated suggestion. If you prefer it the other way, we leave it like that. 👍🏼

> I just copied the snackbar element from somewhere else. I can create a custom snackbar element to streamline the usage if you want

We'll discuss this with UX and change it then. You may leave it for now.